File: core/src/jvmMain/java/de/haukesomm/sokoban/core/level/SokobanLevelCharacterMap.java
package de.haukesomm.sokoban.core.level;

import de.haukesomm.sokoban.core.EntityType;
import de.haukesomm.sokoban.core.TileType;

import java.util.Optional;

public class SokobanLevelCharacterMap implements LevelCharacterMap {

    @Override
    public TileType getTileType(char character) {
        return switch (character) {
            case '#' -> TileType.WALL;
            case '.' -> TileType.TARGET;
            default -> TileType.NOTHING;
        };
    }

    @Override
    public Optional<EntityType> getEntityType(char character) {
        return switch (character) {
            case '@' -> Optional.of(EntityType.PLAYER);
            case '$' -> Optional.of(EntityType.BOX);
            default -> Optional.empty();
        };
    }
}

File: core/src/commonMain/kotlin/de/haukesomm/sokoban/core/level/SokobanLevelCharacterMap.kt
package de.haukesomm.sokoban.core.level

import de.haukesomm.sokoban.core.EntityType
import de.haukesomm.sokoban.core.TileType

class SokobanLevelCharacterMap : LevelCharacterMap {

    override fun getTileType(character: Char): TileType? {
        return when (character) {
            '#' -> TileType.WALL
            '.' -> TileType.TARGET
            else -> TileType.NOTHING
        }
    }

    override fun getEntityType(character: Char): EntityType? {
        return when (character) {
            '@' -> EntityType.PLAYER
            '$' -> EntityType.BOX
            else -> null
        }
    }
}
